lib/src/protocol_instance.rs

```diff
@@ -151,12 +154,12 @@ pub fn assemble_protocol_instance(
             sender: input.taiko.block_proposed.meta.sender,
         },
         prover: input.taiko.prover_data.prover,
-        chain_id: chain_spec.chain_id,
-        sgx_verifier_address: chain_spec.sgx_verifier_address.unwrap_or_default(),
+        chain_id: input.chain_spec.chain_id,
+        sgx_verifier_address: input.chain_spec.sgx_verifier_address.unwrap_or_default(),
     };
 
     // Sanity check
-    if input.network.is_taiko() {
+    if input.chain_spec.is_taiko() {
```

Comment:
Why do we still need `is_taiko` when assemble_protocol_instance only make sense if we're proving taiko?

Reply:
No real reason I think, just minimizing the differences between proving Ethereum and Taiko blocks. Perhaps for Ethereum blocks we can just not call `assemble_protocol_instance` at all which could also minimize the difference.